li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/common/LargeMessageHeaderValue.java

```diff
@@ -81,6 +95,11 @@ public static LargeMessageHeaderValue fromBytes(byte[] bytes) {
     int segmentNumber = PrimitiveEncoderDecoder.decodeInt(bytes, byteOffset);
     byteOffset += PrimitiveEncoderDecoder.INT_SIZE;
     int numberOfSegments = PrimitiveEncoderDecoder.decodeInt(bytes, byteOffset);
-    return new LargeMessageHeaderValue(type, new UUID(mostSignificantBits, leastSignificantBits), segmentNumber, numberOfSegments);
+    if (bytes.length == 29) {
```

Comment:
nit: Can we make this 29 a macro or something? Later when we debug or need to revisit the code, I might forgot what this 29 is.

Reply:
sg! will fix it.